File: packages/client_react/src/types/norm.ts
export interface NormDto {
  id: number;
  name: string;
  description: string | null;
  isObsolete: boolean;
  dateCreated: string;
  dateObsolete: string | null;
  commentsObsolete: string | null;
}

export interface CreateNormDto {
  name: string;
  description: string | null;
}

export interface UpdateNormDto {
  name: string;
  description: string | null;
}

File: server/QCLab/Models/Norm.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class Norm
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    public DateTime DateCreated { get; set; }

    public bool IsObsolete { get; set; }

    public DateTime? DateObsolete { get; set; }

    public string? CommentsObsolete { get; set; }

    public virtual ICollection<Material> Materials { get; set; } = new List<Material>();

    public virtual ICollection<Test> Tests { get; set; } = new List<Test>();
}
